### DISCUSSION TITLE
consider removing `builtins.postcss` support

### DISCUSSION THREAD

**Discussion by hyf0** [web-infra-dev/rspack#3850] [Core] [answered]
`builtins.postcss` is a workaround for not supporting postcss-loader. Now `postcss-loader` is supported, we could remove this feature.

**Comment by hardfist** [comment]
please add the impacts and migration way for users?

**Reply by hyf0** [reply]
impacts: users who use this function.

migration way: switch to `postcss-loader`.

**Comment by zoolsher** [comment]
can we print a warning and also translate the configuration to `module.rules`?

**Reply by hyf0** [reply]
The warning is added. I'm not sure about the printing config part. We could add the migration config in the release note.

**Comment by hyf0** [comment, marked as answer]
Done https://github.com/web-infra-dev/rspack/pull/3849
